internal/pkg/cosign/tsa/client/client.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"bytes"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/digitorus/timestamp"
+	"github.com/pkg/errors"
+)
+
+// TimestampAuthorityClient provides a client to query RFC 3161 timestamp authorities
+type TimestampAuthorityClient struct {
+	Timestamp TimestampAuthorityService
+
+	// URL is the path to the API to request timestamp responses
+	URL string
+
+	// CertificateChain contains an optional certificate chain used to verify timestamp responses
+	CertificateChain []*x509.Certificate
```

Comment:
where does this get populated? it seems like it's not used outside of tests and the flags aren't used to populate it

Is this worth creating a `NewTimestampAuthorityClient` that can populate it?

Reply:
In my opinion, this should be populated when signing to verify the response.

Reply:
This is gone now! Only needed for mocks